Add NCrunch support to weavers by default

ltrzesniewski opened the pull request:
This makes the addins play nicely with NCrunch by default.

I also inverted netclassicweaver/netstandardweaver because I noticed my "fulll .NET" MSBuild was using the netstandardweaver version.

I renamed  `RuntimeToken` to `WeaverRuntimeToken` to make it more specific to Fody, just in case.

Finally, I couldn't reproduce any issues by eliminating the `FilesToAdd` intermediate item, so I removed the indirection.

SimonCropp commented:
Did u confirm that items are being passed in to the fody task, ie by looking at the build log in diognostic mode

ltrzesniewski commented:
I can see the following line in the diagnostic logs after I've modified the PropertyChanged.Fody.props file in my local nuget cache:

```
8>  Task Parameter:WeaverFiles=C:\Dev\dotnet\lib\packages\PropertyChanged.Fody.2.5.12\build\..\netstandardweaver\PropertyChanged.Fody.dll (TaskId:482)
```

This is using the old csproj format.

ltrzesniewski commented:
Wait, that `netstandardweaver` is not good. I'll take a look.

ltrzesniewski commented:
Nevermind, I blame caching for this, as I've edited the props file in the cache. It works fine after restarting VS.

```
1>  Task Parameter:WeaverFiles=C:\Dev\dotnet\lib\packages\PropertyChanged.Fody.2.5.12\build\..\netclassicweaver\PropertyChanged.Fody.dll (TaskId:41)
```

I've also checked the binary log from the build, which shows the correct path:
![image](https://user-images.githubusercontent.com/7913492/46403979-3f43ba80-c704-11e8-9f8b-a689efa2877b.png)

ltrzesniewski commented:
One last thing I'm not sure about: `MSBuildRuntimeType` can also have the `Mono` value. Should it use the classic or standard weaver in that case?

ltrzesniewski commented:
I'm really sorry for the spam, but it just occurred to me there may be a better approach for adding NCrunch support to weavers. I'll try it in a few hours, please don't merge yet.

SimonCropp commented:
lol. people contributing is not spam :)  will hold off

ltrzesniewski commented:
I was referring to that bunch of messages I posted that started to feel like a monologue 😉 

Anyway, I made the change I wanted: I moved the items needed for NCrunch to Fody.targets, because it feels much better that way - there's less logic in the weaver props.

I also added the xmlns to the weaver props because of #560.

Finally, I didn't do anything about the `'$(MSBuildRuntimeType)' == 'Mono'` case I asked about, as the logic used by the weavers is the same that Fody.targets uses which is what matters I think.

Unfortunately, you'll still have to update all addins because netclassicweaver and netstandardweaver were inverted. 😕

ltrzesniewski commented:
BTW there was a disadvantage of using `FilesToAdd`: when using multiple weavers, each one would add an item to `FilesToAdd` and then append all `FilesToAdd` items to `WeaverFiles`.

Here's an example with 3 weavers:

![image](https://user-images.githubusercontent.com/7913492/46462490-e2a6d500-c7c1-11e8-9c40-3c57a4447c2c.png)

The task would get duplicate weaver inputs.

Getting rid of `FilesToAdd` fixes this issue:

![image](https://user-images.githubusercontent.com/7913492/46462452-cc007e00-c7c1-11e8-839c-bdda0a056d93.png)

SimonCropp commented:
thanks heaps for this

ltrzesniewski commented:
@SimonCropp I'm not sure if you missed this detail or if you didn't have the time to do it yet, but since you didn't acknowledge it, I'd like to remind you that *all* addins need to be updated to reference the latest Fody version, because the `netclassicweaver`/`netstandardweaver` directories were inverted.

I guess you must have a script for that 🙂

SimonCropp commented:
Done

ltrzesniewski commented:
Thanks! 👍
